node-scripts/src/pipeline3/impl/dockerComposeGenericMaven.js

```diff
@@ -0,0 +1,454 @@
+  for (const composeFile of composeFiles) {
+    dockerComposeFiles = dockerComposeFiles + ` -f ${composeFiles} `;
+  }
+  return dockerComposeFiles;
+};
+/**
+ * Implementation of script utils to specifically manipulate Docker Compose containers.
+ *
+ */
+module.exports = {
+  preHostPreparation: {
+    getDeploymentScript: function(instanceDef) {
+      // Retrieve the  Docker Compose project
+      const mavenProject = instanceDef.deployment.value.mavenProject;
+      const dockerDirPath = path.resolve(
+        config.getCDDockerDirPath(instanceDef.uuid)
+      );
+      var script = "";
+      script += scripts.initFolder(dockerDirPath, "jenkins", "jenkins", true);
+      script += scripts.fetchArtifact(
+        mavenProject,
+        "maven",
+        dockerDirPath,
+        instanceDef.deployment.value.mavenUrl
+      );
+      return script;
+    },
+    getDataScript: function(instanceDef) {
+      let script = "";
+
+      return script;
+    },
+    getArtifactsScript: function(instanceDef) {
+      return "";
+    }
+  },
+  hostPreparation: {
+    getDeploymentScript: function(instanceDef) {
+      const scripts = require("../scripts");
+      let script = "";
+
+      // Rsync the  Docker Compose project files to the target machine
+      const hostDir = path.resolve(
+        instanceDef.deployment.hostDir,
+        instanceDef.name
+      );
+      const ssh = instanceDef.deployment.host.value;
+      const hostArtifactsDir = hostDir + "/docker_compose";
+      script += scripts.rsync(
+        { ...ssh, ...{ remoteDst: true } },
+        config.getCDDockerDirPath(instanceDef.uuid),
+        hostArtifactsDir,
+        true,
+        null,
+        "-avzz --delete"
+      );
+
+      // Set the Timezone via a env var "TIMEZONE"
+      if (instanceDef.deployment.timezone) {
+        script += scripts.remote(
+          instanceDef.deployment.host.value,
+          scripts.writeProperty(
+            "TIMEZONE",
+            instanceDef.deployment.timezone,
+            path
+              .resolve(
+                instanceDef.deployment.hostDir,
+                instanceDef.name,
+                "docker_compose",
+                ".env"
+              )
+              .toString()
+          )
+        );
+      }
+
+      script += "\n";
+      script += scripts.remote(
+        instanceDef.deployment.host.value,
+        scripts.createEnvVarFileDockerGeneric(instanceDef)
+      );
+      script += "\n";
+
+      // docker-compose build
+      script += scripts.remote(
+        instanceDef.deployment.host.value,
+        "cd " +
+          path
+            .resolve(
+              instanceDef.deployment.hostDir,
+              instanceDef.name,
+              "docker_compose"
+            )
+            .toString() +
+          " && " +
+          composeExec(instanceDef.deployment.composePlugin) +
+          " -p " +
+          instanceDef.name +
+          combineComposeFiles(instanceDef.deployment.dockerComposeFiles) +
```

Comment:
For some reason, it ends up with a command like:
```
docker-compose -p ozone-analytics -f docker-compose-db.yaml,docker-compose-data-pipelines.yaml  -f docker-compose-db.yaml,docker-compose-data-pipelines.yaml  --env-file=/var/docker-volumes/ozone/ozone-analytics/ozone-analytics.env pull
```

Instance Def block:
```
      "dockerComposeFiles": [
        "docker-compose-db.yaml",
        "docker-compose-data-pipelines.yaml"
      ],
```

So I'm not sure what's going on. All the more reasons to add tests to this.